File: app/src/main/java/com/jossy/android/mobilebenchmarkappjava/RAMTestActivity.java
package com.jossy.android.mobilebenchmarkappjava;

import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import android.content.Intent;
import com.jossy.android.mobilebenchmarkappjava.data.TestResult;

public class RAMTestActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_benchmark);

        Log.d("RAMTestActivity", "Starting RAM test");
        startRAMTest();
    }

    private void startRAMTest() {
        long ramStart = System.currentTimeMillis();
        RAMTest.runBenchmark();
        long ramElapsed = System.currentTimeMillis() - ramStart;

        Log.i("RAMTestActivity", "RAM test time: " + ramElapsed + "ms");
        TextView textView = new TextView(this);
        textView.setText("Test RAM ended. Time: " + ramElapsed + " ms");
        setContentView(textView);

        TestResult result = new TestResult("RAM Test", ramElapsed, "Memory allocation test completed", true);
        Intent intent = new Intent();
        intent.putExtra(BenchmarkApplication.RESULT, result);
        setResult(RESULT_OK, intent);
        finish();
    }
}

File: app/src/main/java/com/jossy/android/mobilebenchmarkappkotlin/RAMTestActivity.kt
package com.jossy.android.mobilebenchmarkappkotlin

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.jossy.android.mobilebenchmarkappkotlin.data.TestResult

class RAMTestActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_benchmark)

        Log.d("RAMTestActivity", "Starting RAM test")
        startRAMTest()
    }

    private fun startRAMTest() {
        val ramStart = System.currentTimeMillis()
        RAMTest.runBenchmark()
        val ramElapsed = System.currentTimeMillis() - ramStart

        Log.i("RAMTestActivity", "RAM test time: ${ramElapsed}ms")
        val textView = TextView(this)
        textView.text = "Test RAM ended. Time: $ramElapsed ms"
        setContentView(textView)

        val result = TestResult("RAM Test", ramElapsed, "Memory allocation test completed", true)
        val intent = Intent()
        intent.putExtra(BenchmarkApplication.RESULT, result)
        setResult(RESULT_OK, intent)
        finish()
    }
}
